compose.yaml

```diff
@@ -0,0 +1,121 @@
+x-backend: &backend_base
+  container_name: backend
+  build:
+    context: ./backend
+    dockerfile: backend.containerfile
+  image: backend:latest
+  volumes:
+    - ./backend/tests/ocpperf_test.toml:/opt/backend/ocpperf.toml:Z
+  environment:
+    - CORS_ALLOWED_ORIGINS=http://localhost:3000,localhost:3000,http://frontend:3000,frontend:3000
+
+x-frontend: &frontend_base
+  container_name: frontend
+  build:
+    context: ./frontend
+    dockerfile: frontend.containerfile
+  image: frontend:latest
+
+networks:
+  e2e-pod:
+    driver: bridge
+
+services:
+  searchdb-ocp:
+    container_name: searchdb-ocp
+    image: docker.io/opensearchproject/opensearch:latest
+    healthcheck:
+      test: ["CMD-SHELL", "curl -sSf 'http://localhost:9200/_cluster/health?wait_for_status=yellow&timeout=5s' > /dev/null"]
+      interval: 10s
+      retries: 5
+      start_period: 15s    
+    networks:
+       - e2e-pod
+    ports:
+      - "9200"
+    volumes:
+      - ./backend/tests/functional/setup/ocp_opensearch.yml:/usr/share/opensearch/config/opensearch.yml:z
+      - ./backend/tests/fixtures/search_db_snapshots/ocp-snapshot.tar.gz:/var/tmp/snapshot.tar.gz:z
+    environment:
+      - discovery.type=single-node
+      - DISABLE_INSTALL_DEMO_CONFIG=true
+      - DISABLE_SECURITY_PLUGIN=true
+    entrypoint:
+      - /bin/bash
+      - -c
+      - |
+        # unpack the tar before starting Opensearch
+        cd /var/tmp
+        tar xfz snapshot.tar.gz
+        # now exec the upstream entrypoint so ES boots normally
+        exec /usr/share/opensearch/opensearch-docker-entrypoint.sh      
+
+  db-seed:
+    container_name: db-seed
+    image: functional:latest
+    build:
+      context: ./backend
+      dockerfile: tests/functional.containerfile
+    depends_on:
+      searchdb-ocp:
+        condition: service_healthy
+    networks: [e2e-pod]
+    entrypoint: ["python3", "-u", "tests/db_seed.py"]
+    restart: "no"
+
+  backend:
+    <<: *backend_base
+    depends_on:
+      - searchdb-ocp
+    networks:
+      e2e-pod:
+        aliases: [backend]
+    ports:
+      - "127.0.0.1:8000:8000"
+
+  backend-host:
+    <<: *backend_base
+    profiles: [host]
+    network_mode: host
+    ports:
+      - "8000:8000"
+
+  frontend:
+    <<: *frontend_base
+    build:
+      context: ./frontend
+      dockerfile: frontend.containerfile
+      args:
+        VITE_PERF_DATA_API_BASE_URL: http://backend:8000
+    networks:
+      e2e-pod:
+        aliases: [frontend]
+    ports:
+      - "127.0.0.1:3000:3000"
+
+  frontend-host:
+    <<: *frontend_base
+    profiles: [host]
+    entrypoint: ["nginx", "-g", "daemon off;"]
```

Comment:
Interesting ... this whole `<<:` syntax is new to me and interesting; but in specific right here, why are you overriding the `entrypoint.sh` to drop the DNS check and go directly into NGINX startup? Was the logic not working with `host` networking, for some reason??? 🤔

Reply:
oh yeah that was new to me; docker compose v2 used `extends` for inheritance, but that was deprecated in v3, instead they leverage yaml merge, `<<` for composition. 

We need two different network setups for the e2e tests that are isolated on their own network, and interactive development that uses the host's network stack, so I used the merge stuff to stay dry.

Reply:
`host` networking works, but the frontend's [entrypoint.sh relies on dns being able to resolve backend as a host](https://github.com/mfleader/cpt-dashboard/blob/75cfd83e74e81931ac25b997b58376c95a648e83/frontend/entrypoint.sh#L9); [extra_hosts](https://docs.docker.com/reference/compose-file/build/#extra_hosts) makes it possible for the frontend's entrypoint to resolve that dns entry to where it actually is in our local "host" deployment